### ISSUE TITLE
Restoring checkpoint hangs with multiple threads

### ISSUE THREAD

**Issue by hmenke** [espressomd/espresso#1960]
Just when you thought it was over...

Here is my (not so) minimal script:
```python
from __future__ import print_function
import numpy as np
import espressomd
import espressomd.checkpointing

checkpoint = espressomd.checkpointing.Checkpointing(checkpoint_id="checkpoint")

start_sample = 0
checkpoint.register("start_sample")

system = espressomd.System(box_l = [10,10,10])
system.periodicity = [1,1,1]
system.cell_system.skin = 0.1
system.time_step = 0.01
checkpoint.register("system")

if not checkpoint.has_checkpoints():
    for i in range(10):
        system.part.add(pos=system.box_l * np.random.random(3))
    checkpoint.register("system.part")

if not checkpoint.has_checkpoints():
    checkpoint.save(checkpoint_index=0)

restarted = False
if checkpoint.has_checkpoints():
    print("Restoring Equilibration Checkpoint")
    checkpoint.load(checkpoint_index=0)
    print("Checkpoint loaded")
    restarted = True

for k in range(start_sample,200):
    print(k)
    system.integrator.run(steps=5000, reuse_forces=restarted)
    restarted = False
    start_sample = k + 1
    checkpoint.save(checkpoint_index=0)
```
Reproducible in the Docker container with `maxset.hpp`.
```
root@009346a1bf7b:/tmp/espresso/build# git show
commit 39846e4328e25aa8316bd1d2648a3c9dae8692a0
Merge: 0aad12e e4d7bb4
Author: Florian Weik <florianweik@gmail.com>
Date:   Thu Apr 26 15:00:31 2018 +0200

    Merge pull request #1958 from KaiSzuttor/fix_integrator_checkpoint
    
    integrator: setstate and getstate methods.

root@009346a1bf7b:/tmp/espresso/build# mpirun --allow-run-as-root -n 2 ./pypresso test.py 
0
1
2
3
4
5
6
^Croot@009346a1bf7b:/tmp/espresso/build# mpirun --allow-run-as-root -n 2 ./pypresso test.py 
Restoring Equilibration Checkpoint
```
---

It runs fine in a single thread but I noticed that after restoring the checkpoint the integration seems to be much slower than before.
![test](https://user-images.githubusercontent.com/1903556/39339135-2ad5eb7e-4a1b-11e8-8c7b-9c2ad9d35415.gif)

**Comment by RudolfWeeber** [comment]
There is at least an mpi deadlock in the setter for gamma_rot. Fix
will come some time today.

**Comment by hmenke** [comment]
What is the status?

**Comment by fweik** [comment]
Fix was in #1962, which was merged today.
